python/Ganga/GPIDev/Lib/Job/Job.py

```diff
@@ -1472,6 +1471,9 @@ def submit(self, keep_going=None, keep_on_fail=None, prepare=False):
             # split into subjobs
             rjobs = self._doSplitting()
 
+            if not rjobs or len(rjobs) == 1 and rjobs[0] is self and self.splitter is not None:
```

Comment:
In a lot of cases you will get a list containing only a reference to the single master job and in this case the exception is not thrown as you would expect.
We could fix the rjobs stuff but frankly I'd rather not go near something in-case I break it.